purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -394,24 +437,29 @@ class Purchases internal constructor(
      *
      * @param [activity] Current activity
      * @param [storeProduct] The StoreProduct of the product you wish to purchase
-     * @param [upgradeInfo] The upgradeInfo you wish to upgrade from, containing the oldProductId and the optional
+     * @param [upgradeInfo] The upgradeInfo you wish to upgrade from, containing the oldSku and the optional
      * prorationMode. Amazon Appstore doesn't support changing products so upgradeInfo is ignored for Amazon purchases.
      * @param [listener] The PurchaseCallback that will be called when purchase completes.
      */
+    @Deprecated(
+        "Use purchase() and PurchaseParams.Builder instead",
+        ReplaceWith("purchase()")
+    )
     fun purchaseProduct(
         activity: Activity,
         storeProduct: StoreProduct,
         upgradeInfo: UpgradeInfo,
         listener: ProductChangeCallback
     ) {
-        startProductChange(
-            activity,
-            // TODOBC5 Move this logic to StoreProduct
-            storeProduct.defaultOption?.purchasingData ?: storeProduct.purchasingData,
-            null,
-            upgradeInfo,
-            listener
-        )
+        upgradeInfo.prorationMode?.let { googleProrationMode ->
+            // TODO how to best migrate old proration mode to new?
+//            val rcProrationMode = GoogleProrationMode(googleProrationMode)
```

Comment:
this is an interesting one... should we allow them to pass any proration mode into the old purchase functions, as to not push a breaking change?